lib/pbench/test/unit/server/query_apis/test_datasets_owner.py

```diff
@@ -0,0 +1,249 @@
+from http import HTTPStatus
+from typing import Iterator
+
+import elasticsearch
+import pytest
+
+from pbench.server import JSON
+from pbench.server.database.models.datasets import Dataset
+from pbench.test.unit.server.headertypes import HeaderTypes
+
+
+class TestDatasetOwner:
+    """
+    Unit testing for DatasetOwner class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def fake_elastic(self, monkeypatch, map: JSON, partial_fail: bool):
```

Comment:
Hmmmm...this will be the _third_ copy of this code...should we be trying to share it with the other uses?